redis-snapshot-replicator/functions/shipper.py

```diff
@@ -1,28 +1,20 @@
 import boto3
 import botocore
-import datetime
-import re
 import os
 
-iam = boto3.client('iam')  
-instances = os.environ['DB_INSTANCES']  
 target_bucket = os.environ['TARGET_BUCKET'] 
 
 print('Loading function')
 
-def byTimestamp(snap):  
-    if 'SnapshotCreateTime' in snap:
-        return datetime.datetime.isoformat(snap['SnapshotCreateTime'])
-    else:
-        return datetime.datetime.isoformat(datetime.datetime.now())
-
 def lambda_handler(event, context): 
 
     source = boto3.client('elasticache')
 
-    for instance in instances.split(','):
-        source_snaps = source.describe_snapshots(CacheClusterId=instance)['Snapshots']
-        source_snap = sorted(source_snaps, key=byTimestamp, reverse=True)[0]['SnapshotName']
+    message = event['Records'][0]['Sns']['Message']
```

Comment:
Is it possible that the lambda gets triggered with a batch of sns events? Or is it just a single event per lambda invocation? Asking because I see that `Records` is an array and could potentially have multiple records in it.

Reply:
umm not sure I got what you mean, `[0]` selects the latest (top) record out of them.
triggering is per event but from the documentation it seems lambda can still access all records 

check sample lambdas from AWS for more information: https://docs.aws.amazon.com/lambda/latest/dg/with-sns-create-package.html

Reply:
what I mean is that if there's the possibility that a single lambda invocation contains multiple SNS events. In such case you should iterate over all the records to process them all, no?

Reply:
ah I got you, actually no it's an invocation per each event (I could see this in the debugging data)

in all cases I'll still keep an eye after implementation to make sure it correctly gets triggered and copied on the manual snapshot schedule